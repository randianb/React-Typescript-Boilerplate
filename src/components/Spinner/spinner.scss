.Spinner {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(255, 255, 255, 0.8);
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;
  &.Spinner--noOverlay {
    background: transparent;
  }
}
.Spinner-spinner {
  position: relative;
  display: inline-block;
  width: 28px;
  height: 28px;
  margin-right: 10px;
  vertical-align: middle;
  svg {
    animation-duration: 750ms;
    animation: Spinner-rotate 1s linear infinite;
  }
  circle {
    fill: transparent;
    stroke-width: 4px;
    stroke-dasharray: 128px;
    stroke-dashoffset: 82px;
    stroke: #000;
  }
}
.Spinner-content {
  font-size: 1rem;
  line-height: 1.5;
  color: #000;
  display: inline-block;
}

@-webkit-keyframes Spinner-rotate {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes Spinner-rotate {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
