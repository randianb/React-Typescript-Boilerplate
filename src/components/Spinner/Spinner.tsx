import React from 'react';
import classNames from 'classnames';
import './spinner.scss';

export interface SpinnerDisplayProps {
  text?: string;
  noOverlay?: boolean;
  className?: any;
}

class Spinner extends React.Component<SpinnerDisplayProps> {
  public static defaultProps = {
    text: '',
    noOverlay: false
  };

  public render() {
    const { text, noOverlay, className, ...rest } = this.props;

    const classes = classNames(
      'Spinner',
      noOverlay && 'Spinner--noOverlay',
      className
    );

    return (
      <div {...rest} className={classes}>
        <div className="Spinner-spinner">
          <svg viewBox="0 0 64 64">
            <circle transform="translate(32,32)" r="26" />
          </svg>
        </div>
        {text ? <div className="Spinner-content">{text}</div> : null}
      </div>
    );
  }
}

export default Spinner;
