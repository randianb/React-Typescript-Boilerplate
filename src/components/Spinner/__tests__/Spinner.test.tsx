import React from 'react';
import Spinner from '../Spinner';
import { render, cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('Spinner', () => {
  it('Renders Spinner', () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders Spinner without overlay', () => {
    const { container } = render(<Spinner noOverlay={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders Spinner with text', () => {
    const { container } = render(<Spinner text="Foo" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
