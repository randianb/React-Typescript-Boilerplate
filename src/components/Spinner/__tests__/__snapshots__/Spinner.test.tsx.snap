// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Renders Spinner 1`] = `
<div
  class="Spinner"
>
  <div
    class="Spinner-spinner"
  >
    <svg
      viewBox="0 0 64 64"
    >
      <circle
        r="26"
        transform="translate(32,32)"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner Renders Spinner with text 1`] = `
<div
  class="Spinner"
>
  <div
    class="Spinner-spinner"
  >
    <svg
      viewBox="0 0 64 64"
    >
      <circle
        r="26"
        transform="translate(32,32)"
      />
    </svg>
  </div>
  <div
    class="Spinner-content"
  >
    Foo
  </div>
</div>
`;

exports[`Spinner Renders Spinner without overlay 1`] = `
<div
  class="Spinner Spinner--noOverlay"
>
  <div
    class="Spinner-spinner"
  >
    <svg
      viewBox="0 0 64 64"
    >
      <circle
        r="26"
        transform="translate(32,32)"
      />
    </svg>
  </div>
</div>
`;
